packages/xo-server/src/api/network.mjs

```diff
@@ -73,12 +80,25 @@ export async function set({
 }) {
   network = this.getXapiObject(network)
 
+  let nbdPromise
+
+  if (nbd !== undefined) {
+    nbdPromise = new Promise(resolve => {
+      if (nbd) {
+        if (network.purpose.includes('insecure_nbd')) network.remove_purpose('insecure_nbd')
+        resolve(network.add_purpose('nbd'))
+      } else {
+        resolve(network.remove_purpose('nbd'))
+      }
+    })
+  }
+
```

Comment:
I think it can be more readable using an async function . 
```suggestion
  async function updateNbd(nbd){
    if(nbd){
      if (network.purpose.includes('insecure_nbd')){
        await network.remove_purpose('insecure_nbd')
      }
      await network.add_purpose('nbd')
      resolve()
    } else {
      network.remove_purpose('nbd')
    }
  }
```